### PR TITLE
fixes #9030 - Adds support to clone config template via api

### PR DESCRIPTION
**Pull request by shlomizadok**

### CODE REVIEW

**Comment by orrabin** [comment]
@shlomizadok tested this and it works well

File: app/controllers/api/v2/config_templates_controller.rb [hunk 1]
```diff
@@ -84,6 +84,23 @@ def build_pxe_default
...
+        end
+      end
+
+      api :POST, "/config_templates/:id/clone", N_("Clone a provision template")
+      param :id, :identifier, :required => true
+      param_group :config_template_clone, :as => :create
+
+      def clone
+        @config_template = @config_template.clone
+        load_vars_for_form
```
Comment (ohadlevy) [inline app/controllers/api/v2/config_templates_controller.rb]: nitpick: name is misleading, as there is no api form.

File: app/controllers/concerns/foreman/controller/config_templates.rb [hunk 2]
```diff
@@ -1,6 +1,15 @@
 module Foreman::Controller::ConfigTemplates
   extend ActiveSupport::Concern
 
+  def load_vars_for_form
+    return unless @config_template
+
+    @locations = @config_template.locations
```
Comment (ohadlevy) [inline app/controllers/concerns/foreman/controller/config_templates.rb]: nitpick: ident based on '='

File: test/functional/api/v2/config_templates_controller_test.rb [hunk 3]
```diff
@@ -74,4 +74,20 @@ class Api::V2::ConfigTemplatesControllerTest < ActionController::TestCase
...
     assert_equal "aha", config_templates(:pxekickstart).audits.last.comment
   end
+
+  test 'should clone template' do
+    original_config_template = config_templates(:pxekickstart)
+    post :clone, { :id => original_config_template.to_param,
+                   :config_template => {:name => 'MyClone'} }
+    assert_response :success
+    template = ActiveSupport::JSON.decode(@response.body)
+    assert(template['name'], 'MyClone')
```
Comment (ohadlevy) [inline test/functional/api/v2/config_templates_controller_test.rb]: assert_equal

File: app/controllers/api/v2/config_templates_controller.rb (hunk 1 above)
Comment (shlomizadok) [inline app/controllers/api/v2/config_templates_controller.rb]: actually using the same method of the web ui.
Will refactor anyhow.

**Comment by ohadlevy** [comment]
merged as fa94105 thanks @shlomizadok

**Pull request closed without merging**